Add support for calling commands by attr access

miedzinski opened the pull request:
Adds support for calling rlite commands by accessing attributes in `hirlite.hirlite.Rlite`.

Until `hirlite.hirlite.Rlite` gets more attributes than `command`, we can make the assumption that by accessing any attribute, we mean calling command.

Unfortunately, I'm not too familiar with writing C extensions -- `hirlite.Rlite` is now written in Python and is a subclass of `hirlite.hirlite.Rlite`. C doesn't support currying functions, so in order to achieve this we should either use `functools.partial` from C (which gives us nothing over this solution) or modify `rlite.c` a bit more.

Additionally, I have changed relative imports to absolute.

miedzinski committed:
Add support for calling commands by attr access

seppo0010 commented:
Can you add a test that uses this new code?

miedzinski committed:
Redirect 'delete' to 'del' in Rlite.__getattr__

miedzinski committed:
Add tests for Rlite.__getattr__

miedzinski commented:
Done.

Additionally, in order to use `DEL` you have to call `Rlite.delete` (just like in redis-py).

miedzinski commented:
I see that Travis build fails -- but it's test runner's fault. I have pushed another patch.

edit:
Looks like test runner requires a bit more work. It is relying too much on editing `sys.path` and relative imports, which isn't good practice. :)
I have deleted last commit -- fixing test runner should be in another pull request. I will try to fix it when I get back home.
New feature itself is working and you can prove it by executing this:

```
python setup.py build
python setup.py develop
cd test
python -m unittest rlite.RliteTest.test_getattr rlite.RliteTest.test_getattr_err
```

seppo0010 commented:
Thanks for the help. I did my best, but my python is not great. I'll wait for the other PR to be in so ci is not broken. Thanks again.

miedzinski committed:
Fix relative imports and travis config

Under py2, without __future__, we were still using
relative imports -- fixed.

Travis CI build relied on relative imports as well.
Its install command was just building C extension and
the tested module was hirlite's source code instead of
installed package. Now, Travis CI properly installs
hirlite and runs test.py.

In test.py, instead of adding build/ to sys.path, we are
*removing* working directory from sys.path --
if we didn't do that, wrong package was imported.

miedzinski commented:
See 0849715995183639d58e7d72595cb4f0b3d8d932 message for explanation. All tests are passing now.
